src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -379,6 +380,42 @@ object GenomeReference {
     gr
   }
 
+  def importReferences(hConf: Configuration, path: String) {
+    if (hConf.exists(path)) {
+      val refs = hConf.listStatus(path)
+      refs.foreach { fs =>
+        val grPath = fs.getPath.toString
+        val gr = hConf.readFile(grPath) { (is: InputStream) => fromJSON(JsonMethods.parse(is)) }
+        val name = gr.name
+        if (!GenomeReference.hasReference(name))
+          addReference(gr)
+        else {
+          if (GenomeReference.getReference(name) != gr)
+            fatal(s"`$name' already exists and is not identical to the imported reference from `$grPath'.")
+        }
+      }
+    }
+  }
+
+  private def writeReference(hc: HailContext, path: String, gr: GRBase) {
+    val grPath = path + gr.name + ".json.gz"
+    if (!hailReferences.contains(gr.name) && !hc.hadoopConf.exists(grPath))
+      gr.asInstanceOf[GenomeReference].write(hc, grPath)
+  }
+
+  def exportReferences(hc: HailContext, path: String, t: Type) { (t: @unchecked) match {
```

Comment:
can this write the same reference multiple times?

Reply:
No, `writeReference` is guarded, and only writes if the file doesn't already exist.